integration_test/third_party_apps_data/applications/cassandra/debian_ubuntu/install

```diff
@@ -3,12 +3,17 @@ set -e
 curl https://downloads.apache.org/cassandra/KEYS | sudo apt-key add -
 
 source /etc/os-release
-if [[ "${VERSION_ID}" == 23* ]]; then
+if [[ "${VERSION_ID}" == 23* || "$(uname -m)" == aarch64 ]]; then
     # There is a python dependency that fails for any older versions of cassandra on Ubuntu 23.04, so explicitly install Cassandra 4.1x.
+    # Additionally, cassandra doesn't have arm64 binaries for 2.2x, but does for 4.1x.
     echo "deb https://debian.cassandra.apache.org 41x main" | sudo tee -a /etc/apt/sources.list.d/cassandra.sources.list
 
     sudo apt update
-    sudo apt install -y openjdk-11-jdk cassandra
+
+    # cassandra doesn't run on Java 17, but on some platforms (debian-11-arm64) it has a transitive dependency on Java 17 (???).
+    # Installing openjdk-11-jdk first ensures that it takes precedence over Java 17 on PATH.
```

Comment:
This isn't supposed to be how Debian handles conflicting binary names... each package that provides a binary (like `java`) has a priority, and every time you install or remove one of those packages it chooses the highest-priority version out of the currently installed ones.

Are you sure it actually has a transitive dependency on Java 17, and not just a dependency on a JRE that apt is resolving to Java 17 by default? If it's the latter, you can just specify both package names on a single `apt install`.

Reply:
I'm not a fan of this code either, but putting them in a single `apt install` simply doesn't work. Cassandra just picks whatever JRE is on PATH, which ends up being Java 17 when installed this way, which it doesn't support:

```
$ sudo apt install -y openjdk-11-jdk cassandra
$ /usr/sbin/cassandra
OpenJDK 64-Bit Server VM warning: Option UseBiasedLocking was deprecated in version 15.0 and will likely be removed in a future release.
Unrecognized VM option 'UseConcMarkSweepGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
$ java --version
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment (build 17.0.7+7-Debian-1deb11u1)
OpenJDK 64-Bit Server VM (build 17.0.7+7-Debian-1deb11u1, mixed mode, sharing)
```

If I do `apt depends cassandra` then I get:

```
$ sudo apt depends cassandra                                                                                                                       
cassandra                                                                                                                                                                   
 |Depends: <openjdk-8-jre-headless>                                                                                                                                         
 |Depends: <java8-runtime>                                                                                                                                                  
    default-jre                                                                                                                                                             
    openjdk-11-jre                                                                                                                                                          
    openjdk-17-jre                                                                                                                                                          
 |Depends: openjdk-11-jre-headless
  Depends: <java11-runtime>
    default-jre
    openjdk-11-jre
    openjdk-17-jre
...
```

Now, if I do `apt install openjdk-11-jre-headless cassandra` instead, _then_ it works. So there's something incorrect about using `openjdk-11-jdk` I guess?